components/script/dom/bindings/global.rs

```diff
@@ -175,10 +176,10 @@ impl<'a> GlobalRef<'a> {
 
     /// `ScriptChan` used to send messages to the event loop of this global's
     /// thread.
-    pub fn user_interaction_task_source(&self) -> Box<ScriptChan + Send> {
+    pub fn user_interaction_task_source(&self) -> Box<TaskSource<UserInteractionTask> + Send> {
```

Comment:
Why though? Is it because workers are not supposed to have the concept of task sources at all? If so, I might just remove all other task sources for `GlobalRoot`.